runtime/src/startup/mod.rs

```diff
@@ -85,21 +85,12 @@ extern "C" fn rust_start() -> ! {
         static rt_header: RtHeader;
     }
 
-    // TODO: Implement a safe memop API in libtock_platform and migrate these
-    // calls to that API.
-    //
-    // Safety: Memop operations 10 and 11 are always memory-safe, as they do not
-    // impact the execution of this process.
     #[cfg(not(feature = "no_debug_memop"))]
+    // Safety: rt_header is defined in the linker script, valid for its type,
+    // and not modified anywhere
     unsafe {
-        TockSyscalls::syscall2::<{ syscall_class::MEMOP }>([
-            10u32.into(),
-            rt_header.stack_top.into(),
-        ]);
-        TockSyscalls::syscall2::<{ syscall_class::MEMOP }>([
-            11u32.into(),
-            rt_header.initial_break.into(),
-        ]);
+        let _ = TockSyscalls::memop_debug_stack_start(rt_header.stack_top as *const u8);
+        let _ = TockSyscalls::memop_debug_heap_start(rt_header.initial_break as *const u8);
```

Comment:
```
error[E0277]: the trait bound `*const u8: From<*mut ()>` is not satisfied
  --> runtime/src/startup/mod.rs:92:75
   |
92 |         let _ = TockSyscalls::memop_debug_stack_start(rt_header.stack_top.into());
   |                                                                           ^^^^ the trait `From<*mut ()>` is not implemented for `*const u8`
   |
   = help: the following other types implement trait `From<T>`:
             <u8 as From<bool>>
             <u8 as From<NonZeroU8>>
   = note: required for `*mut ()` to implement `Into<*const u8>`
```

It does not work because of the intermediate type that was introduced by my helper function